cmd/snap/cmd_run.go

```diff
@@ -705,6 +739,10 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 		logger.Noticef("WARNING: cannot copy user Xauthority file: %s", err)
 	}
 
+	if err := ensureXdgDocumentPortal(info); err != nil {
```

Comment:
Could we tie this to the presence of _something_ in the snap?

Reply:
Yes.  I was thinking of checking for the presence of a desktop interface plug.  Just haven't written the code yet.  And I also need to add some testing for all of this.

Reply:
The unfortunate part of snap run is that it cannot just go and ask snapd about the connection status but I think in this case it would be okay to just start it iff a plug is present.